### ISSUE TITLE
Avoid or mitigate overfitting in Image categorization using CNN

### ISSUE THREAD

**Issue by jingweimo** [fchollet/keras#4325]
I am working on using CNN to perform image categorization. There are 10 categories of images each of them has about 300-500 images.  During training, it is observed that the training accuracy (close to 90%) is much higher than the validation accuracy (about 30%, only better than random guess), indicating a severe overfitting issue.  I have randomly selected 80% of images per category for training and the rest for validation. 

Is there any  strategies to avoid or mitigate the overfitting issue, and meanwhile to improve the validation performance. I have tried to use early stopping, but the testing accuracy is still very low.

**Comment by varun-bankiti** [comment]
1. Seems you have less training data, try to use **Data Augmentation** techniques  like [this](https://keras.io/preprocessing/image/), [this](https://deeplearningmania.quora.com/The-Power-of-Data-Augmentation-2) and [this] (https://www.quora.com/How-exactly-do-data-augmentation-techniques-like-mirroring-and-cropping-reduce-overfitting)
2. **Make the model simple** - If the model have too many parameters, instead of generalizing it simply memorizes the training data.
3. Use [**Regularization Techniques**](https://en.wikipedia.org/wiki/Regularization_(mathematics)) like [Dropout](https://www.cs.toronto.edu/~hinton/absps/JMLRdropout.pdf), [Regularizers](https://keras.io/regularizers/) - [More about Regularization](https://www.quora.com/What-is-regularization-in-machine-learning)

Good Luck (y)
